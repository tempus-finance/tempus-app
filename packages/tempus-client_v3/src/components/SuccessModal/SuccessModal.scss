.tc__success-modal__content {
  max-width: 480px;
  padding: 0 24px 48px 24px;
  text-align: center;
}

.tc__success-modal__animation {
  margin: 24px 0;
  position: relative;

  .tc__success-modal__title {
    position: absolute;
    left: 50%;
    top: 50%;
    transform: translate(-50%, -50%);
  }
}

.tc__success-modal__action-buttons {
  display: flex;
  flex-direction: column;
  align-items: center;

  .tc__actionButton:first-child {
    margin-bottom: 16px;
  }
}
