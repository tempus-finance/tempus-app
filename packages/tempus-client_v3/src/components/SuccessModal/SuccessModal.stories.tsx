import { ComponentStory, ComponentMeta } from '@storybook/react';
import { useCallback, useState } from 'react';

import SuccessModal from './SuccessModal';

export default {
  title: 'SuccessModal',
  component: SuccessModal,
  argTypes: {},
} as ComponentMeta<typeof SuccessModal>;

const style = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  height: '100px',
};

const Template: ComponentStory<typeof SuccessModal> = () => {
  const [modalOpen, setModalOpen] = useState(false);

  const toggleModal = () => {
    setModalOpen(prevState => !prevState);
  };

  const onModalClose = useCallback(() => {
    setModalOpen(false);
  }, []);

  return (
    <div style={style}>
      <button type="button" onClick={toggleModal}>
        Click me!
      </button>

      <SuccessModal
        open={modalOpen}
        onClose={onModalClose}
        title="Deposit Complete!"
        description="You have deposited 0.5 ETH with 10/10/2022 term. It should reach your wallet momentarily. "
        primaryButtonLabel={{ default: 'Deposit in another pool', loading: '', success: '' }}
        onPrimaryButtonClick={() => {}}
        secondaryButtonLabel={{ default: 'Manage Portfolio', loading: '', success: '' }}
        onSecondaryButtonClick={() => {}}
      />
    </div>
  );
};

export const Primary = Template.bind({});
