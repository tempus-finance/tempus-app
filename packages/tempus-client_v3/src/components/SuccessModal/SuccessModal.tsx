import Lottie from 'lottie-react';
import { FC, memo } from 'react';
import { ActionButton, ActionButtonLabels, Modal, Typography } from '../shared';
import { ModalProps } from '../shared/Modal/Modal';
import confettiAnimation from './animation/confetti.json';
import './SuccessModal.scss';

export interface SuccessModalProps extends ModalProps {
  description: string;
  primaryButtonLabel: ActionButtonLabels;
  onPrimaryButtonClick: () => void;
  secondaryButtonLabel: ActionButtonLabels;
  onSecondaryButtonClick: () => void;
}

const SuccessModal: FC<SuccessModalProps> = props => {
  const {
    open,
    onClose,
    title,
    description,
    primaryButtonLabel,
    onPrimaryButtonClick,
    secondaryButtonLabel,
    onSecondaryButtonClick,
  } = props;

  return (
    <Modal open={open} onClose={onClose}>
      <div className="tc__success-modal__content">
        <Typography variant="body-primary">{description}</Typography>
        <div className="tc__success-modal__animation">
          <Lottie animationData={confettiAnimation} loop={false} autoplay />
          <Typography className="tc__success-modal__title" variant="subtitle" weight="bold">
            {title}
          </Typography>
        </div>
        <div className="tc__success-modal__action-buttons">
          <ActionButton
            labels={secondaryButtonLabel}
            onClick={onSecondaryButtonClick}
            variant="secondary"
            size="large"
          />
          <ActionButton labels={primaryButtonLabel} onClick={onPrimaryButtonClick} variant="primary" size="large" />
        </div>
      </div>
    </Modal>
  );
};

export default memo(SuccessModal);
