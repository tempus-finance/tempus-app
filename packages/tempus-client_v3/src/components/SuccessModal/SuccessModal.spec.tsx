import { fireEvent, render } from '@testing-library/react';
import SuccessModal, { SuccessModalProps } from './SuccessModal';

const mockOnPrimaryButtonClick = jest.fn();
const mockOnSecondaryButtonClick = jest.fn();

const defaultProps: SuccessModalProps = {
  open: true,
  onClose: () => {},
  title: 'Success',
  description: 'Modal description',
  primaryButtonLabel: {
    default: 'Primary Button',
    loading: '',
    success: '',
  },
  onPrimaryButtonClick: mockOnPrimaryButtonClick,
  secondaryButtonLabel: {
    default: 'Secondary Button',
    loading: '',
    success: '',
  },
  onSecondaryButtonClick: mockOnSecondaryButtonClick,
};

jest.mock('lottie-react', () => () => <div className="lottie-animation" />);

const subject = (props: SuccessModalProps) => render(<SuccessModal {...props} />);

describe('SuccessModal', () => {
  it('renders a success modal', () => {
    const { container, getByText } = subject(defaultProps);

    const animationContainer = container.querySelector('.tc__success-modal__animation .lottie-animation');
    const title = getByText(defaultProps.title as string);

    expect(animationContainer).not.toBeNull();
    expect(title).not.toBeNull();

    expect(container).not.toBeNull();
    expect(container).toMatchSnapshot();
  });

  it('calls `onClick` when primary button is clicked', async () => {
    const { getByRole } = subject(defaultProps);

    const primaryButton = getByRole('button', { name: defaultProps.primaryButtonLabel.default });

    expect(primaryButton).not.toBeNull();
    expect(primaryButton).toHaveClass('tc__actionButton-border-primary-large');

    fireEvent.click(primaryButton);

    expect(mockOnPrimaryButtonClick).toHaveBeenCalled();
    expect(primaryButton).toMatchSnapshot();
  });

  it('calls `onClick` when secondary button is clicked', () => {
    const { getByRole } = subject(defaultProps);

    const secondaryButton = getByRole('button', { name: defaultProps.secondaryButtonLabel.default });

    expect(secondaryButton).not.toBeNull();
    expect(secondaryButton).toHaveClass('tc__actionButton-border-secondary-large');

    fireEvent.click(secondaryButton);

    expect(mockOnSecondaryButtonClick).toHaveBeenCalledTimes(1);
    expect(secondaryButton).toMatchSnapshot();
  });
});
