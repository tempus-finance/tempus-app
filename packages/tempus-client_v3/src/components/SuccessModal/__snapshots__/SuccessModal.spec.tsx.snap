// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SuccessModal calls \`onClick\` when primary button is clicked 1`] = `
<button
  class="tc__btn tc__actionButton tc__actionButton-border-primary-large tc__actionButton-background-primary-large tc__actionButton-hover-primary-large tc__actionButton-shadow-default tc__actionButton-cursor-default tc__actionButton-size-large "
  data-selected="false"
  type="button"
>
  <div
    class="tc__actionButton-status-label"
  >
    <div
      style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(255, 255, 255); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      Primary Button
    </div>
  </div>
</button>
`;

exports[`SuccessModal calls \`onClick\` when secondary button is clicked 1`] = `
<button
  class="tc__btn tc__actionButton tc__actionButton-border-secondary-large tc__actionButton-background-secondary-large tc__actionButton-hover-secondary-large tc__actionButton-shadow-default tc__actionButton-cursor-default tc__actionButton-size-large "
  data-selected="false"
  type="button"
>
  <div
    class="tc__actionButton-status-label"
  >
    <div
      style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
    >
      Secondary Button
    </div>
  </div>
</button>
`;

exports[`SuccessModal renders a success modal 1`] = `
<div>
  <div
    class="tc__modal-backdrop"
    id="modal-backdrop"
  />
  <div
    class="tc__modal-container "
  >
    <div
      class="tc__modal tc__modal__plain tc__modal__size-small"
    >
      <div
        class="tc__modal-header"
      >
        <div
          style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
        />
        <button
          class="tc__btn "
          data-selected="false"
          type="button"
        >
          <svg
            class="tc__icon tc__icon-close"
            fill="none"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M15.6081 13.7161C15.859 13.967 16 14.3073 16 14.6621C16 15.017 15.859 15.3572 15.6081 15.6081C15.3572 15.859 15.017 16 14.6621 16C14.3073 16 13.967 15.859 13.7161 15.6081L8 9.87872L2.2839 15.6081C2.033 15.859 1.6927 16 1.33788 16C0.98305 16 0.642756 15.859 0.391855 15.6081C0.140955 15.3572 0 15.017 0 14.6621C0 14.3073 0.140955 13.967 0.391855 13.7161L6.12128 8L0.391855 2.2839C0.140955 2.033 -2.64367e-09 1.6927 0 1.33788C2.64367e-09 0.98305 0.140955 0.642756 0.391855 0.391855C0.642756 0.140955 0.98305 2.64367e-09 1.33788 0C1.6927 -2.64367e-09 2.033 0.140955 2.2839 0.391855L8 6.12128L13.7161 0.391855C13.967 0.140955 14.3073 0 14.6621 0C15.017 0 15.3572 0.140955 15.6081 0.391855C15.859 0.642756 16 0.98305 16 1.33788C16 1.6927 15.859 2.033 15.6081 2.2839L9.87872 8L15.6081 13.7161Z"
              fill="#222222"
            />
          </svg>
        </button>
      </div>
      <div
        class="tc__modal__body"
      >
        <div
          class="tc__success-modal__content"
        >
          <div
            style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 400; font-family: 'DM Sans', sans-serif; opacity: 1;"
          >
            Modal description
          </div>
          <div
            class="tc__success-modal__animation"
          >
            <div
              class="lottie-animation"
            />
            <div
              class="tc__success-modal__title"
              style="font-style: normal; font-size: 20px; line-height: 28px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
            >
              Success
            </div>
          </div>
          <div
            class="tc__success-modal__action-buttons"
          >
            <button
              class="tc__btn tc__actionButton tc__actionButton-border-secondary-large tc__actionButton-background-secondary-large tc__actionButton-hover-secondary-large tc__actionButton-shadow-default tc__actionButton-cursor-default tc__actionButton-size-large "
              data-selected="false"
              type="button"
            >
              <div
                class="tc__actionButton-status-label"
              >
                <div
                  style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(34, 34, 34); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
                >
                  Secondary Button
                </div>
              </div>
            </button>
            <button
              class="tc__btn tc__actionButton tc__actionButton-border-primary-large tc__actionButton-background-primary-large tc__actionButton-hover-primary-large tc__actionButton-shadow-default tc__actionButton-cursor-default tc__actionButton-size-large "
              data-selected="false"
              type="button"
            >
              <div
                class="tc__actionButton-status-label"
              >
                <div
                  style="font-style: normal; font-size: 16px; line-height: 24px; color: rgb(255, 255, 255); font-weight: 700; font-family: 'DM Sans', sans-serif; opacity: 1;"
                >
                  Primary Button
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
